Adds a set of cmdlets for provisioning and manipulating Modern Client…

ypcode opened the pull request:
… Side pages

## Type ##
- [ ] Bug Fix
- [x] New Feature
- [ ] Sample

## What is in this Pull Request ? ##
Adds wrapper Cmdlets for the recent PnP Core Library Modern Pages provisioning features.
It is aligned to the fresh breaking changes in the PnP Core lib dev branch. This should be only be released with the Core library v2.18.1709.0

msftclas commented:
@ypcode,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by Microsoft. We will now review your pull request.
_Thanks,
Microsoft Pull Request Bot_

erwinvanhunen commented:
Great PR, Yannick!

I had to manually merge it due to some conflicts with other changed files. At the same time we decided to refactor your cmdlets a bit, to keep them more in sync with the others. Among others we removed the functionality of the 'in-memory' page. Instead we will only allow updating the page in place. This will eliminate the step to save the page, which is not a common pattern in PowerShell. As a result we removed the New-PnPClientSidePage.

We also refactored the cmdlets a bit to use parametersets, which will force the user to always pick the right set of parameters, eliminating some checks in the cmdlets. This will also show the correct usage scenarios automatically in the help files. Besides that we fixed a few typos and parameter mismatches in the help for the cmdlets.

We also removed the ability to auto generate a page name which we noticed was based upon a static variable you increased. As this is local to the current instance in PowerShell, launching a new instance would have resetted this back to 1, optionally causing conflicts with existing pages. That could potentially cause confusion why it would not be able to auto generate a new page name. 

But again, great functionality and thank you!

ypcode commented:
Glad you appreciate it Erwin, it is a pleasure to contribute!

Sorry for the lot of refactoring you had to do...

Feel free to ask or suggest any addition I could bring to it ;)

________________________________
From: Erwin van Hunen <notifications@github.com>
Sent: Thursday, September 7, 2017 9:08:41 PM
To: SharePoint/PnP-PowerShell
Cc: Yannick Plenevaux; Mention
Subject: Re: [SharePoint/PnP-PowerShell] Adds a set of cmdlets for provisioning and manipulating Modern Client… (#1008)


Great PR, Yannick!

I had to manually merge it due to some conflicts with other changed files. At the same time we decided to refactor your cmdlets a bit, to keep them more in sync with the others. Among others we removed the functionality of the 'in-memory' page. Instead we will only allow updating the page in place. This will eliminate the step to save the page, which is not a common pattern in PowerShell. As a result we removed the New-PnPClientSidePage.

We also refactored the cmdlets a bit to use parametersets, which will force the user to always pick the right set of parameters, eliminating some checks in the cmdlets. This will also show the correct usage scenarios automatically in the help files. Besides that we fixed a few typos and parameter mismatches in the help for the cmdlets.

We also removed the ability to auto generate a page name which we noticed was based upon a static variable you increased. As this is local to the current instance in PowerShell, launching a new instance would have resetted this back to 1, optionally causing conflicts with existing pages. That could potentially cause confusion why it would not be able to auto generate a new page name.

But again, great functionality and thank you!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/SharePoint/PnP-PowerShell/pull/1008#issuecomment-327895118>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AH9X1HYt7-FWmqQlJBV8AbSiZ_N6SJPdks5sgD85gaJpZM4PDEhe>.
